helix-core/src/test/java/org/apache/helix/controller/stages/TestIntermediateStateCalcStage.java

```diff
@@ -248,26 +255,28 @@ public void testWithClusterConfigChange() {
 
     IntermediateStateOutput output = event.getAttribute(AttributeName.INTERMEDIATE_STATE.name());
 
-    for (String resource : resourceSet) {
+    for (String resource : resources) {
       // Note Assert.assertEquals won't work. If "actual" is an empty map, it won't compare
       // anything.
-      Assert.assertTrue(output.getPartitionStateMap(resource).getStateMap()
-          .equals(expectedResult.getPartitionStateMap(resource).getStateMap()));
+      Assert.assertEquals(output.getPartitionStateMap(resource).getStateMap(), expectedResult.getPartitionStateMap(resource).getStateMap());
```

Comment:
This is my test script to verify that some partitions do get double-charged, but it seems after the fix, there's no special assertions to claim that these partitions are not in the output?

Reply:
I am not sure whether you understand the scenario. Once the pending messages added to resource0 and the throttling set to 3. p0 and p1 will be charged for pending and only 1 quota leftover. Then for recovery throttle, p0 will take the quota and p1 will be throttled. So it is unexpected result.

With the fix, the expected result will match the computed result.

Reply:
My observations of the test code are that in this final Assertion stage, other than the format change, the content doesn't change at all. It shouldn't be the case because you modified the behavior of adding the pending ST messages, right? Something or some comments need to be here. Otherwise, without adding or adding the ST messages, the test script stays same?
If I missed something, please let me know.